python3. avoid constraint breaks in bsd loaders. use cachecontrol

jayblunt opened the pull request:
This contains:

1: python3

2: try/catch around the bsd loader inserts. the dogma loader populates some of the dgmEffect tables which then causes constraint errors on the load later. 

3: cachecontrol replaces redis_cache in getitems / getgroups.

FrancisTheDutchman commented on requirements.txt:
```diff
@@ -1,2 +1,15 @@
+chardet==3.0.4
+idna==2.9
+msgpack==1.0.0
+pkg-resources==0.0.0
```
this is a non valid thing. Have you generated this requirements directly from what you had installed? Recommend to remove this line (have tested this myself)

jayblunt replied:
Which line was invalid?

The requirements.txt was generated directly from pip freeze (python 3.8.2 / pip 20.1.1) I and then added the --with-libyaml global option back for PyYAML.

FrancisTheDutchman replied:
in github the comment is always directly below the line that is mentioned (confusing, I agree), so in this case line 6. pkg-resources is something from the OS I believe and shouldnt be in requirements. also I think the version pinning like this is a little to restrictive, minor updates should not break the script.

jayblunt replied:
This makes sense, thank you. i have updated requirements.txt and removed the version pinning and extra packages. I also took out the --global-option="--with-libyaml" aas well, because it prevents pip from installing wheels.

FrancisTheDutchman commented:
tested it with latest SDE export for sqlite and worked perfectly!

jayblunt commented on tableloader/tableFunctions/bsdTables.py:
```diff
@@ -1,35 +1,37 @@
+                try:
+                    connection.execute(tablevar.insert().values(row))
+                except sqlalchemy.exc.IntegrityError as err:
+                    print("{} skipped {} ({})".format(tablename, row, err))
```
It is noisy. But a constraint violation shouldn't be happening, so until I know what @fuzzysteve wanted with the bsdTables loader overwriting the dogma effects loaders I chose to leave it in.

FrancisTheDutchman replied:
that is a good point, finding out why this occurs. But by printing this this much it has an heavy performance impact due to the terminal not flushing fast enough.

fuzzysteve commented:
I'll have a think about merging this. The upgrade to python 3 is the main reason I'm not just doing it. Would have to upgrade the environment I'm running it in. 
As for the bsd tables, I have a script which wraps this, and deletes the unwanted dogma tables (as well as downloading the sde, shoving it in git, running the getItems, etc etc)

FrancisTheDutchman commented:
would it be possible to share the script for deleting the unwanted dogma tables? Thanks for your awesome work!

fuzzysteve commented:
I've just added runconversion.sh
It's the wrapper I use. It's _very_ specific to my process.

jayblunt commented:
Thanks @fuzzysteve, adding runconversion.sh helps me figure out how you run this. I've adjusted bsdTables.py to skip the dgm* tables and fixed up TypesToJson.py. I'm not running the entire process via runconversion.sh yet - i don't have your tree structure - but all the parts seem to work.

FrancisTheDutchman commented:
hi man nice work! I added a small extra thing as a pullrequest to your fork, packaged volume to invTypes. I thought it was better to start from your fork since it has python3 support

jayblunt commented:
@fuzzysteve looks like tableloader/tableFunctions/typeMaterials.py got missed in the last commit. I have updated the PR but without the new typeMaterials.py I am stuck.

jayblunt commented:
Updated from https://github.com/fuzzysteve/yamlloader/commit/af0dbc8705932fb1bf8bb3821852ac2f827edfc8

jayblunt commented:
(let's see if i can update the branch on this pr)

jayblunt commented:
(nope, looks like I can only update the dst branch, not the src branch)
